validator/src/main/java/com/google/daq/mqtt/util/FileDataSink.java

```diff
@@ -68,7 +68,9 @@ private File getOutputFile(String deviceId, String topic) {
   private File getOutputFile(String deviceId, String subType, String subFolder, String suffix) {
     File deviceDir = getDeviceDir(deviceId);
     deviceDir.mkdirs();
-    return new File(deviceDir, String.format("%s_%s.%s", subType, subFolder, suffix));
+    boolean invalid = subFolder == null || "update".equals(subFolder);
+    return new File(deviceDir,
+        String.format("%s%s.%s", subType, invalid ? "" : "_" + subFolder, suffix));
```

Comment:
Suggestion, use the string format to without additional tiny + string concat in the arguments?

Reply:
Updated to make it, er, less ugly.